src/components/role-review-card.tsx

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @next/next/no-img-element */
+import { cn } from "~/lib/utils";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "~/components/ui/card";
+import { api } from "~/trpc/server";
+import { ReviewCardStars } from "./review-card-stars";
+
+// TODO: move this somewehre else?
+function formatDate(date: Date) {
+  const months = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+
+  // Extract the year, month, and day components
+  const year = date.getFullYear();
+  const monthIndex = date.getMonth();
+  const month = months[monthIndex];
+  const day = String(date.getDate()).padStart(2, "0"); // Ensure two digits for the day
+
+  // Construct the formatted date string
+  const formattedDate = `${day} ${month} ${year}`;
+
+  return formattedDate;
+}
+
+// todo: add this attribution in a footer somewhere
+//  <a href="https://clearbit.com">Logos provided by Clearbit</a>
+
+type RoleReviewCardProps = {
+  className?: string;
+  roleObj: {
+    id: string;
+    title: string;
+    description: string | null;
+    companyId: string;
+  };
+};
+
+export async function RoleReviewCard({
+  className,
+  roleObj,
+  ...props
+}: RoleReviewCardProps) {
+  // ===== COMPANY DATA ===== //
+  const company = await api.company.getById.query({ id: roleObj.companyId });
+  const roleDescription =
+    roleObj.description && roleObj.description.length >= 210
+      ? cn(roleObj.description.slice(0, 210), "...")
+      : roleObj.description;
+  const positionDate: string = formatDate(company.createdAt);
+
+  // ===== REVIEW DATA ===== //
+  const reviews = await api.review.getByRole.query({ roleId: roleObj.id });
+  const reviewCount = reviews.length;
+  let workEnvironment = "In Person";
+  // TODO: we are setting the work enviornment to the first role since its not stored per company. Change?
+  if (reviews.length > 0 && reviews[0]) {
+    if (reviews[0].workEnvironment == "HYBRID") {
+      workEnvironment = "Hybrid";
+    } else if (reviews[0].workEnvironment == "REMOTE") {
+      workEnvironment = "Remote";
+    }
+  }
+
+  // ===== AVERAGE RATING ===== //
+  // TODO: Abstract?
+  const totalStars = reviews.reduce((accum, curr) => {
+    return accum + curr.overallRating;
+  }, 0);
+  const averageStars = totalStars / reviewCount;
+
+  return (
+    <Card
+      className={cn(
+        "flex h-[26rem] w-96 flex-col justify-between overflow-hidden rounded-3xl",
+        className,
+      )}
+      {...props}
+    >
+      <div>
+        <CardHeader className="pb-3">
+          <div className="flex justify-end text-xs font-extralight leading-3">
+            {positionDate}
+          </div>
+          <div className="flex items-center justify-start space-x-4">
+            <img
```

Comment:
Can we change this to `Image` from `next/image`? You may have to modify `next.config.js` since we're using an external URL.

Reply:
Wasn't able to get this working, maybe we can try and figure it out at gm

Reply:
Yes, not a problem! The only advantage of the `<Image />` tag is the optimization and lack of layout shift. We can figure it out at GM.